autopark/Program.cs

```diff
@@ -6,33 +6,62 @@ class Program
+                 new Vehicle(new VehicleType("types[0]", 1.2F), new ElectricalEngine(50), "Electric Bus E321", "6785 BA-7", 12080, 2019, 20451, Color.Green, 150),
+                 new Vehicle(new VehicleType("types[1]", 1F), new DiedelEngine(1.6, 7.2), "Golf 5", "8682 AX-7", 1200, 2006, 230451, Color.Gray, 55),
+                 new Vehicle(new VehicleType("types[1]", 1F), new ElectricalEngine(25), "Tesla Model S 70 D", "E001 AA-7", 2200, 2019, 10454, Color.White, 70),
+                 new Vehicle(new VehicleType("types[2]", 1.5F), new DiedelEngine(3.2, 25), "Hamm HD 12 VV", null, 3000, 2016, 122, Color.Yellow, 20),
+                 new Vehicle(new VehicleType("types[3]", 1.2F), new DiedelEngine(4.75, 20.1), "МТЗ Беларус-1025.4", "1145 AB-7", 1200, 2020, 109, Color.Red, 135)
             };
-            float maxTaxCoef = 0, sumTaxCoef = 0, averTaxCoef;
 
             n = vehicles.Length;
-            for (i = 0; i < n; i++) {
-                if (i == n - 1)
-                    vehicles[i].TaxCoefficient = 1.3F;
 
-                vehicles[i].Display();
+            Console.WriteLine("Initial array:");
+            Console.WriteLine(PrintVehicle.ToString(vehicles));
 
-                if (vehicles[i].TaxCoefficient > maxTaxCoef)
-                    maxTaxCoef = vehicles[i].TaxCoefficient;
+            Array.Sort(vehicles);
 
-                sumTaxCoef += vehicles[i].TaxCoefficient;
+            Console.WriteLine("\nSorted array:");
+            Console.WriteLine(PrintVehicle.ToString(vehicles));
+
+            for (i = 0; i < n; i++)
+            {
+                if (vehicles[lowestMiliage].MileageKm > vehicles[i].MileageKm)
+                    lowestMiliage = i;
+
+                if (vehicles[highestMiliage].MileageKm < vehicles[i].MileageKm)
+                    highestMiliage = i;
             }
 
-            Console.WriteLine($"Max tax coefficient is {maxTaxCoef}");
+            Console.WriteLine();
+            Console.WriteLine("Vehicle with the lowest miliage:");
+            Console.WriteLine(vehicles[lowestMiliage].ToString());
+            Console.WriteLine("Vehicle with the highest miliage:");
+            Console.WriteLine(vehicles[highestMiliage].ToString());
+
+
+            Console.WriteLine("\nEqual vehicles are:");
+            bool[] isToPrint = new bool[n];
+            
+            for (i = 0; i < n - 1; i++)
+            {
+                for (j = i + 1; j < n; j++)
+                {
+                    if (vehicles[i].Equals(vehicles[j]))
+                    {
+                        isToPrint[i] = true;
+                        isToPrint[j] = true;
+                    }
+                }
+            }
 
-            averTaxCoef = sumTaxCoef / n;
-            Console.WriteLine($"Average tax coefficient is {averTaxCoef}");
+            for (i = 0; i < n; i++)
+                if (isToPrint[i])
+                    Console.WriteLine(vehicles[i].ToString());
```

Comment:
Just advice: don't use inline constructions because it's difficult to read and looks like ugly...

Reply:
![image](https://user-images.githubusercontent.com/79978383/116143274-9c992000-a6e3-11eb-9d7e-264b07eaed5a.png)
Done